.github/workflows/nightly.yml

```diff
@@ -0,0 +1,151 @@
+name: Nightly
+on:
+  # FIXME: Change to run on a schedule before merging
+  #schedule:
+  #  - cron: "0 1 * * *"
+  pull_request:
+    branches:
+      - '**'
+  workflow_dispatch: ~
+  issue_comment:
+    types: [created]
+jobs:
+  GENERATE_MATRIX:
+    name: Generate Matrix
+    if: ${{ github.event_name != 'issue_comment' || (github.event.issue.pull_request && contains(github.event.comment.body, '@php run-extended-tests')) }}
+    runs-on: ubuntu-latest
+    outputs:
+      branches: ${{ steps.set-matrix.outputs.branches }}
+      asan-matrix: ${{ steps.set-matrix.outputs.asan-matrix }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          # Set fetch-depth to 0 to clone the full repository
+          # including all branches. This is required to find
+          # the correct commit hashes.
+          fetch-depth: 0
+      - name: Grab the commit mapping
+        uses: actions/cache@v3
+        with:
+          path: branch-commit-cache.json
+          # The cache key needs to change every time for the
+          # cache to be updated after this job finishes.
+          key: nightly-${{ github.run_id }}-${{ github.run_attempt }}
+          restore-keys: |
+            nightly-
+      - name: Generate Matrix
+        id: set-matrix
+        run: php .github/nightly_matrix.php "${{ github.event_name }}" "${{ github.run_attempt }}" "${{ github.event.pull_request.head.sha }}"
```

Comment:
Not sure if the `"${{ github.event.pull_request.head.sha }}"` parameter will cause issues for non-PR events (due to a missing object key). You'll probably need to test after merging.

Reply:
I wasn't sure about that either. We'll see ^^ Thanks for reviewing

Reply:
I tried the comment trigger in my fork and it didn't work, so I removed it for now.